### PR TITLE
Class Modules (#690)

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Solve Issue #690 

```python
from src.taipy.gui import Markdown, Gui, Page

class TestModule(Page):
    def __init__(self):
        self.renderer = Markdown(
            """
x = <|{x}|>

<|Set x to 30|button|on_action=set_x|>
"""
        )
        self.x = 10
        self.y = 20
        self.register_page()

    def set_x(self):
        self.x = 30

    def on_change(self):
        pass


gui = Gui()
gui.add_page("test", TestModule())
gui.run(run_browser=False)
```

### CODE REVIEW

File: src/taipy/gui/page.py [hunk 1]
```diff
@@ -30,66 +31,74 @@ class Page(ABC):
...
-        else:
-            self._content = content
-
-    def set_content(self, content: str) -> None:
-        """Set a new page content.
-
-        Reads the new page content and reinitializes the page to reflect the change.
-
-        !!! important
-            This function can only be used an IPython notebook context.
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/page.py:73]: used an => used in an

File: src/taipy/gui/renderers/__init__.py [hunk 2]
```diff
@@ -10,23 +10,88 @@
...
+        Arguments:
+            content (str): The text content or the path to the file holding the text to be transformed.
+
+        If *content* is a path to a readable file, the file is read entirely as the text template.
+        """
+        super().__init__(**kwargs)
+        content: str = kwargs.get("content", None)
+        if content is None:
+            raise ValueError("'content' argument is missing for class '_Renderer'")
+        if not isinstance(content, str):
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/renderers/__init__.py:37]: should we accept a pathlib.Path ?

File: src/taipy/gui/renderers/__init__.py [hunk 3]
```diff
@@ -10,23 +10,88 @@
...
+        else:
+            self._content = content
+
+    def set_content(self, content: str) -> None:
+        """Set a new renderer content.
+
+        Reads the new renderer content and reinitializes the _Renderer instance to reflect the change.
+
+        !!! important
+            This function can only be used an IPython notebook context.
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/renderers/__init__.py]: in an

**Review by FredLL-Avaiga — APPROVED** [review]

File: src/taipy/gui/renderers/__init__.py (hunk 3 above)
Comment (FabienLelaquais) [reply src/taipy/gui/renderers/__init__.py]: Thx

**Comment by FabienLelaquais** [comment]
The syntax to use:

```py
from taipy.gui import Gui, Page, Markdown

class MainPage(Page):
    
    def __init__(self):
        self.v = 0
        super().__init__()

    def create_page(self):
        return Markdown("""<|{v}|slider|on_change=slider_changed|>""")

    def slider_changed(state, var_name, var_value):
        print(f"slider has changed: {var_value}")

gui = Gui(page=MainPage()).run(run_browser=False)
```

This technique does NOT however access the class variables.

**Comment by dinhlongviolin1** [comment]
> The syntax to use:
> 
> ```python
> from taipy.gui import Gui, Page, Markdown
> 
> class MainPage(Page):
>     
>     def __init__(self):
>         self.v = 0
>         super().__init__()
> 
>     def create_page(self):
>         return Markdown("""<|{v}|slider|on_change=slider_changed|>""")
> 
>     def slider_changed(state, var_name, var_value):
>         print(f"slider has changed: {var_value}")
> 
> gui = Gui(page=MainPage()).run(run_browser=False)
> ```
> 
> This technique does NOT however access the class variables.

This now work
```
from src.taipy.gui import Gui, Page, Markdown

class MainPage(Page):
    
    v = 30

    def create_page(self):
        return Markdown("""<|{v}|slider|on_change=slider_changed|>""")

    def slider_changed(state, var_name, var_value):
        print(f"slider has changed: {var_value}")

gui = Gui(page=MainPage()).run(run_browser=False)
```

**Review by FabienLelaquais — COMMENTED** [review]
👍

**Pull request merged**